frontend/src/store/FamLoginUserState.ts

```diff
@@ -0,0 +1,168 @@
+    const accesses = getUserAccess();
+    if (accesses && accesses.length > 0) {
+        const access = accesses.filter(
+            (access) => access.auth_key == adminRole
+        )[0]
+        return access
+            .grants
+            .map((grant) => grant.application)
+            .sort((first, second) => first.id - second.id);
+    }
+};
+
+/**
+ * Note!! this is to combine all applications the user has been granted
+ * and can administer on with different admin level privileges, only for
+ * special usage at dashboard page (ManagePermissions) since the user when
+ * it lands on the first page it needs to select "application" first.
+ *
+ * If the way our forntend works is based on admin level to display
+ * management table instead of selecting "applicatoin" first, it won't need
+ * this particular function.
+ */
+const getApplicationsUserAdministers = () => {
+    const userAdminRoleGroups = getUserAdminRoleGroups();
+    let famApp;
+    const applicationList: Array<FamApplicationDto> = [];
+    if (userAdminRoleGroups) {
+        userAdminRoleGroups.forEach(adminRoleGroup => {
+            const apps = getApplicationsAdministeredByAdminRole(adminRoleGroup)
+            if (adminRoleGroup == AdminRoleAuthGroup.FamAdmin) {
+                famApp = apps!.filter((app) => app.name == FAM_APPLICATION_NAME)[0];
+            }
+            else {
+                apps?.forEach((app) => {
+                    let isNewItem = true
+                    for (let item of applicationList) {
+                        if (item.id == app.id) {
+                            isNewItem = false;
+                            break;
+                        }
+                    }
+                    if (isNewItem) applicationList.push(app);
+                });
+            }
+        });
+
+        applicationList.sort((first, second) => first.id - second.id);
+        if (famApp) applicationList.unshift(famApp); // add FAM to the first if FAM Admin.
+    }
+    return applicationList;
+};
+
+const hasAccessRole = (role: string): boolean => {
+    if (state.value.famLoginUser?.roles?.includes(role)) {
+        return true;
+    }
+    return false;
+};
+
+// --- setters
+
+const storeFamUser = (famLoginUser: FamLoginUser | null | undefined) => {
+    state.value.famLoginUser = famLoginUser;
+    if (famLoginUser) {
+        localStorage.setItem(
+            FAM_LOGIN_USER,
+            JSON.stringify(state.value.famLoginUser)
+        );
+    } else {
+        localStorage.removeItem(FAM_LOGIN_USER);
+    }
+};
+
+const removeFamUser = () => {
+    storeFamUser(undefined);
+    // clean up local storage for selected application
+    localStorage.removeItem(CURRENT_SELECTED_APPLICATION_KEY);
+};
+
+/**
+ * To refresh User granted privileges when neccssary.
+ * FamLoginUser contains a property "accesses" that sometimes needs to be
+ * refreshed and stored into localStorage at FamLoginUser object.
+ */
+const refreshCachedUserAccess = async () => {
```

Comment:
maybe we shouldn't call it refreshCached anymore, cause we only call it when we login, and there won't be any cached access